src/embedded/classes/IterableEmbeddedManager.test.ts

```diff
@@ -167,5 +167,92 @@ describe('IterableEmbeddedManager', () => {
     });
   });
 
+  describe('startImpression', () => {
+    it('should call IterableApi.startEmbeddedImpression with messageId and placementId', () => {
+      // GIVEN a message ID and placement ID
+      const messageId = 'message-123';
+      const placementId = 456;
+
+      // WHEN startImpression is called
+      embeddedManager.startImpression(messageId, placementId);
+
+      // THEN IterableApi.startEmbeddedImpression is called with the correct parameters
+      expect(MockRNIterableAPI.startEmbeddedImpression).toHaveBeenCalledTimes(
+        1
+      );
+      expect(MockRNIterableAPI.startEmbeddedImpression).toHaveBeenCalledWith(
+        messageId,
+        placementId
+      );
+    });
+
+    it('should handle multiple impression starts', () => {
+      // GIVEN multiple messages
+      const messageId1 = 'message-1';
+      const placementId1 = 100;
+      const messageId2 = 'message-2';
+      const placementId2 = 200;
+
+      // WHEN startImpression is called multiple times
+      embeddedManager.startImpression(messageId1, placementId1);
+      embeddedManager.startImpression(messageId2, placementId2);
```

Comment:
Not exactly remembering how the session works. If mutiple embedded messages can gather in one session and how impressions are tracked. What should starting a session mean if it is called twice. Does it append over active session?